### DISCUSSION TITLE
[zh_TW] General Translations Fix

### DISCUSSION THREAD

**Discussion by KogentaSan** [OpenTTD/team#579] [Other]
@OpenTTD/zh_tw 

As this is a problem related to languages, Traditional Chinese will be used in this discussion.

問題
----- 
現時OpenTTD的繁體中文翻譯有很多不足的地方：

- 1a. 語句沒有被翻譯
- 2a. 翻譯的語句過時
- 3a. 非標準繁體中文字元（導致遊戲亂碼不能自動轉用繁體中文字體的問題）
- 4a. 現時的翻譯詞語及標點符號格式標準不一導致混亂
- 5a. 現時所翻譯的文章句子讀起上來很不流暢

目標
-----

- 1b. 翻譯欠缺的部份
- 2b. 修改過時翻譯的語句
- 3b. 修改非標準繁體中文字元到標準繁體中文字元
- 4b. 統一翻譯詞語及標點符號格式
- 5b. 修飾文章語法使到語句讀起來通順流暢

實行
-----

- [x] 1c. 翻譯欠缺的部份已經被翻譯了
- [x] 2c. 過時的翻譯語句亦已被修改
- [x] 3c. 非標準繁體中文字元經已被修正，現時測試版在遊戲中選擇繁體中文時不再出現亂碼，
　 亦不再需要修改openttd.cfg內的字體檔案位置都能正確顯示字體
- [ ] 4c. 有待實行。如能得到討論共識，就需要對現時的翻譯內容作出全面的修改
- [ ] 5c. 有待實行。如能得到討論共識，就需要對現時的翻譯內容作出全面的修改

後記
-----
本次發文希望能明確定下一個實行的方向，令將來加入的翻譯者有一個明確要做的目標。
僅一人之力能做的事真的很有限，希望能夠集液成球，多些人過來幫忙翻譯及討論今後要做的事吧。
另外也期望能使用國語的台灣人來幫忙修飾文法，
因本人並非使用國語為母語的香港人，
在語句或用詞上難免與使用國語為母語的人有很大的分別呢。

**Comment by wensimehrp** [comment]
@KogentaSan 
虽然并非和原版 OpenTTD 直接相关，有计划改进 JGRPP 的翻译吗？目前发现有一部分台湾玩家使用简体中文而非繁体中文，虽然事实上有助于玩家间交流，但是归根到底，简中不能替代繁中；简中与台湾繁中的语句词语文字差异在一些情况下会导致理解困难。

另外还有一点：是否应该拆分 zh_TW 与 zh_HK。<https://github.com/OpenTTD/team/discussions/528>。

**Reply by KogentaSan** [reply]
關於JGRPP的翻譯問題，目前官方的Web翻譯中並未涵蓋JGRPP，所以目前無法提供幫助。
我個人只能在翻譯方面提供有限的幫助，並且無法在程式碼層面進行更改。
至於將zh_TW和zh_HK拆分的問題，這涉及到主程式的架構，
需要在另一個結構層面的討論區進行討論。
而有關上次拆分的討論亦因zh_TW的翻譯不足及人手不足而終止討論了，參見 [#9801](https://github.com/OpenTTD/OpenTTD/pull/9801)，
但現在zh_TW的翻譯已有顯著改進，
可能是時候重新開啟這方面的討論了，
建議您在另一邊討論區開設新主題，與相關權限人士討論。

**Reply by wensimehrp** [reply]
> 關於JGRPP的翻譯問題，目前官方的Web翻譯中並未涵蓋JGRPP，所以目前無法提供幫助。
> 我個人只能在翻譯方面提供有限的幫助，並且無法在程式碼層面進行更改。
> 至於將zh_TW和zh_HK拆分的問題，這涉及到主程式的架構，
> 需要在另一個結構層面的討論區進行討論。
> 而有關上次拆分的討論亦因zh_TW的翻譯不足及人手不足而終止討論了，參見 [#9801](https://github.com/OpenTTD/OpenTTD/pull/9801)，
> 但現在zh_TW的翻譯已有顯著改進，
> 可能是時候重新開啟這方面的討論了，
> 建議您在另一邊討論區開設新主題，與相關權限人士討論。

事实上 JGRPP 的翻译文本目前是通过 Pull request 提交的；我可以提供提交方面的帮助。除此之外，JGRPP 的简体中文翻译相当完善……可以以此为蓝本。

**Reply by KogentaSan** [reply]
JGRPP的繁體翻譯以簡體為本的轉換已初步成功了
但還是需要再多的時間修飾一下用詞
另外因字體的問題「U+200B ZERO WIDTH SPACE (零寬空格)」不知會否出現
還有就是會否有非標準繁體字在裡面而引致遊戲發生亂碼的情況
這些都需要更多的時間找尋出來才能提交翻譯呢。

另外官方穩定版的繁體亂碼問題需要等待下一次穩定版更新才能解決，
有說要下年25年4月後才有機會有穩定版的更新。

**Reply by wensimehrp** [reply]
@KogentaSan 

我用繁体中文没有任何问题（Manjaro Linux / Windows 11 Simplified Chinese 都试过了）乱码是因为没有找到合适的字体文件。“官方稳定版的乱码问题需要等下一次更新才可以解决”是从哪里得来的消息呢？（是不是 OpenTTD/OpenTTD#13064 ?）

至于 `U+200B`，游戏在编译时会自行检查是否有此类字符。如果有则会在编译时提示，实际上很好解决，如果实在没有条件自行编译也可以依赖 github 的自动编译检查（jgrpp 和 openttd 的 pull request 会自动运行编译检查）或者自己写一个脚本检查。

非标准繁体字我不清楚是什么情况，但是我觉得游戏并不会因此产生严重乱码。

https://github.com/OpenTTD/OpenTTD/issues/12738 -> 无合适的字体文件导致的乱码\
https://github.com/OpenTTD/OpenTTD/issues/12881 -> 有 `U+200B` 导致编译不通过\
https://github.com/OpenTTD/OpenTTD/issues/13064 -> 之前你开的 issue，可能被理解为“游戏语言文件更新”问题而不是“繁体中文乱码”问题。我刚才测试并没有乱码。
https://github.com/OpenTTD/OpenTTD/issues/7567 -> 乱码问题，已解决

**Reply by KogentaSan** [reply]
JGRPP繁中那邊大致已翻譯完成，只餘個別用字選擇上有些猶豫，但基本已經可用且暢通，問題不大。
現在反而是官方程式這邊還未開始修飾用詞及句子，這邊的問題會比較複雜一點，因為既有的翻譯已很早存在，
在考慮是否選用原有字詞及語句上已經有很多的糾結，要考慮到選用台灣用語或港式書面語已經很傷腦筋，
因現時只有一種繁體中文，在選詞上得選一些兩者都看得明的用詞及用語，但還是真的很難，
加上個人並不對台灣用語十分了解，只得多在網路上尋找資料作校對才能確定。
耐何現時台灣翻譯組的人並不活躍，想作更多討論及落實還得要多待一會兒了。

[#12738](https://github.com/OpenTTD/OpenTTD/issues/12738) -> 我記得當時在午夜構建測試版的簡中、日文、韓文都解決問題的時候，就只有繁中還是照樣出現一堆(????)的問號字體顯示，直至我將內文有香港增補字體修改做標準字體後就一切正常顯示了。

[#13064](https://github.com/OpenTTD/OpenTTD/issues/13064) -> 情況有點不一樣，你試想如果一個翻譯檔裡有多國語言如繁體加簡體再加日文漢字，而那個系統字形檔案就是沒有那些綜合各國漢字字形，結果系統就是配對不到字型，那個舊翻譯內容就是那麼的糟糕。
再加上繁中亂碼的問題都已經發生很久了，看看Steam的評論都是自2023年3月起的，我也只是想能正常玩到才尋根究底的來這裡幫忙翻譯的。 : -(
另外就個人感覺而言，繁中Windows系統就是寫得不完全不完整，不像簡中Windows系統的相容性來得高一點。

**Reply by ahyangyi** [reply]
感觉这个开发中的 Pull Request 可能对过去一段时间各种 “fallback 导致的字体/字型问题” 有所帮助。

https://github.com/OpenTTD/OpenTTD/pull/13303
